pkg/services/secrets/secrets.go

```diff
@@ -42,3 +44,16 @@ type Provider interface {
 	Encrypt(ctx context.Context, blob []byte) ([]byte, error)
 	Decrypt(ctx context.Context, blob []byte) ([]byte, error)
 }
+
+type ProviderID string
+
+func (id ProviderID) Kind() (string, error) {
+	idStr := string(id)
+
+	parts := strings.SplitN(idStr, ".", 2)
+	if len(parts) != 2 {
+		return "", fmt.Errorf("malformatted provider identifier %s: expected format <provider>.<keyName>", idStr)
```

Comment:
Doesn’t this break backwards compatibility?

Reply:
In theory, [these changes](https://github.com/grafana/grafana/pull/43640/commits/ee0ed8fba31ed6886b2ff0de11c3faa52c6fb674) should fix it. WDYT? 🤔